network/src/validation.rs

```diff
@@ -263,27 +244,26 @@ pub struct NetworkDown;
 
 /// A future that resolves when a collation is received.
 pub struct AwaitingCollation {
-	outer: futures::sync::oneshot::Receiver<::futures::sync::oneshot::Receiver<Collation>>,
-	inner: Option<::futures::sync::oneshot::Receiver<Collation>>
+	outer: oneshot::Receiver<oneshot::Receiver<Collation>>,
+	inner: Option<oneshot::Receiver<Collation>>
 }
 
 impl Future for AwaitingCollation {
-	type Item = Collation;
-	type Error = NetworkDown;
+	type Output = Result<Collation, NetworkDown>;
+
+	fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+		let this = Pin::into_inner(self);
```

Comment:
For what it's worth, if you put `mut self: Pin<&mut Self>` instead, then you don't need this line. It's fine either way.

Reply:
🤷‍♀ that's going to get turned into an async function either way